36 tutorial scenes

anthonyargatoff opened the pull request:
Added a basic tutorial that runs through 5 scenes for a beginner player.

## Menu
![image](https://github.com/user-attachments/assets/e3179129-d570-485c-8084-4ee3e0e04936)

## Tutorial 1:
![image](https://github.com/user-attachments/assets/e2f8eb3e-c190-4065-9aa9-92bf955da466)

## Tutorial 2:
![image](https://github.com/user-attachments/assets/fc35ad5d-cb14-4be7-9e39-e670b9f8107c)

## Tutorial 3:
![image](https://github.com/user-attachments/assets/e5e9ea4d-a69e-460a-9b54-4d63af538fde)

## Tutorial 4:
![image](https://github.com/user-attachments/assets/092b6400-3cd9-470a-85c2-74487816276f)

## Tutorial 5:
![image](https://github.com/user-attachments/assets/cee35e2a-3ef1-40ae-8753-1d7995d0668f)
![image](https://github.com/user-attachments/assets/4f8cfef9-8b29-46d4-aa11-7b5939e3f017)

After reaching the flag in tutorial level 5, will redirect the player to the main menu.

A nice improvement would be to add a swipe effect when changing tutorial scenes, instead of the fade.

anthonyargatoff committed:
Make tutorial button

anthonyargatoff committed:
Tutorial scenes 1 and 2. Button goes to tutorial

anthonyargatoff committed:
Add tutorial 3 and 4

anthonyargatoff committed:
Navigation for tutorial

anthonyargatoff committed:
tut 3 changes

anthonyargatoff committed:
Pause menu. Adjust platform hitboxes

bhunt02 reviewed (changes requested):
Looks pretty good, but do you think we could make these a little less wide? I've added a prefab for stage walls that we could constrict and use to make the tutorial space less broad, it's a little jarring at the moment

bhunt02 commented:
Also, I think the scale on the beast is a bit off since it's not in the 2,2,1 scale that the stage environments normally are, so the barrels look really huge (same thing with the barrels, but the scaling is just off in general there)

anthonyargatoff committed:
Change scale tut 1

anthonyargatoff committed:
Change scale tut 2

anthonyargatoff committed:
Scaled tut 3

anthonyargatoff committed:
Tutorial 4 scaling

anthonyargatoff committed:
Change tutorial 5 scaling issues

anthonyargatoff commented:
- Changed the scaling of tutorial levels 1, 2, and 3 to be smaller
- Changed the wheel platform scaling in tutorial 4
- Changed the beast scaling, and the barrel scaling in tutorial 5

## 1
![image](https://github.com/user-attachments/assets/96f01179-a001-4ea7-9bfd-ca5828f673f0)

## 2
![image](https://github.com/user-attachments/assets/623dff90-d472-41b9-8c6b-1b47ee46252d)

## 3
![image](https://github.com/user-attachments/assets/a3573821-1f12-4ef8-a4a4-5675befaac8d)

## 4
![image](https://github.com/user-attachments/assets/439c087d-2aea-45e7-8966-8578adf95095)

## 5
![image](https://github.com/user-attachments/assets/af2dd446-d68b-4322-bda5-3de63849df25)

anthonyargatoff committed:
Update barrel spawner. Set vals to 0

anthonyargatoff committed:
Merge branch 'main' into 36-tutorial-scenes

anthonyargatoff committed:
Adjust barrel sizes

anthonyargatoff committed:
Editor changes

bhunt02 reviewed (approved):
These look good. I'll make adjustments as needed in my finalization branch to ensure everything works together. I will say I already know we'll have to correct the text on the tutorials as they used constant pixel size rather than scaling, but I'm working on that already. Good job!

bhunt02 commented on Assets/Scripts/Entities/Boss/BarrelSpawner.cs:
```diff
@@ -36,6 +37,9 @@ private IEnumerator SpawnBarrels()
         yield return new WaitForSeconds(randomWaitTime);
       }
       
+      if (barrelSize > 0) {
+        barrel.transform.localScale = new Vector3(barrelSize, barrelSize, barrelSize);
+      }
       Barrel newBarrel = Instantiate(barrel, barrelSpawnRef.position, Quaternion.identity);
```
I'll fix this, but this code actually changes the prefab's scale, which isn't great lol

bhunt02 committed:
Merge branch 'main' of https://github.com/anthonyargatoff/MCTSWCFA into 36-tutorial-scenes

# Conflicts:
#	Assets/Fonts/PublicPixel SDF.asset
#	Assets/Scripts/GameManager.cs
#	Assets/Scripts/UI/MainMenuButtons.cs
#	Assets/UI/Rewind/RewindEffect.mat
#	ProjectSettings/EditorBuildSettings.asset
